File: app/src/main/java/com/faendir/lightning_launcher/multitool/proxy/ActivityScreen.java
package com.faendir.lightning_launcher.multitool.proxy;

import android.content.Intent;

/**
 * @author lukas
 * @since 08.07.18
 */
public interface ActivityScreen extends Screen {
    ImageBitmap cropImage(ImageBitmap image, boolean full_size);

    void hideActionBar();

    int pickColor(String title, int color, boolean hasAlpha);

    Image pickImage(int maxPixels);

    float pickNumericValue(String title, float value, String valueType, float min, float max, float interval, String unit);

    void showActionBar(Function onCreateOptionsMenu, Function onOptionsItemSelected);

    boolean startActivityForResult(Intent intent, Script receiver, String token);
}

File: app/src/main/java/com/faendir/lightning_launcher/multitool/proxy/ActivityScreen.kt
package com.faendir.lightning_launcher.multitool.proxy

import android.content.Intent

/**
 * @author lukas
 * @since 08.07.18
 */
interface ActivityScreen : Screen {
    fun cropImage(image: ImageBitmap, full_size: Boolean): ImageBitmap

    fun hideActionBar()

    fun pickColor(title: String, color: Int, hasAlpha: Boolean): Int

    fun pickImage(maxPixels: Int): Image

    fun pickNumericValue(title: String, value: Float, valueType: String, min: Float, max: Float, interval: Float, unit: String): Float

    fun showActionBar(onCreateOptionsMenu: Function, onOptionsItemSelected: Function)

    fun startActivityForResult(intent: Intent, receiver: Script, token: String): Boolean
}
